src/logic/Applications.swift

```diff
@@ -126,6 +126,10 @@ class Applications {
     private static func notAltTab(_ app: NSRunningApplication) -> Bool {
         return app.processIdentifier != ProcessInfo.processInfo.processIdentifier
     }
+    
+    private static func isPasswords(_ app: NSRunningApplication) -> Bool {
+        return app.bundleIdentifier == "com.apple.Passwords"
```

Comment:
Hi,

Thank you for sharing this workaround.

Could you please share which checks are not passing for the Passwords app currently?

I'm a bit worried to bypass the checks like this. Maybe this app is not supposed to be monitored by accessibility apps?

Thank you 🙇

Reply:
So Passwords.app is an XPC app making windows? This seems new and surprising.

[This guide](https://multi.app/blog/how-and-when-to-use-xpc-services) says:

> XPC services have one major limitation: they do not accept user interaction. This may be a deal breaker for you. You can use an external process to show UI. The external process can have its own window hierarchy and use SwiftUI, but the user cannot interact with anything in the external process. UI elements such as buttons or sliders will not accept mouse or keyboard events.

We see here that [yabai also started filtering XPC out](https://github.com/koekeishiya/yabai/issues/529#issue-620346055). They [still have that check](https://github.com/koekeishiya/yabai/blob/136f25f515d07ff941231fa34a4fbea2448ca95e/src/process_manager.c#L40) today.

It seems that it's not even possible to use AppKit from inside an XPC app. Maybe they use tricks like have an app inside an app, and inheriting the execution context?

I'm worried to hard-code an exception like this. What do you think about an XPC app producing windows?

Thank you 🙇